util/codec/codec.go

```diff
@@ -833,6 +879,34 @@ func DecodeOne(b []byte) (remain []byte, d types.Datum, err error) {
 	return b, d, nil
 }
 
+// DecodeIndexOne decodes on datum from []byte of `KindMysqlTime`.
+func DecodeIndexOne(b []byte) (remain []byte, d types.Datum, err error) {
```

Comment:
How about renaming it to `DecodeAsDateTime`?

Reply:
I agree with you!